src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -227,6 +229,152 @@ def statfs(self, path):
             'avail_bytes': statfs.avail_blocks * statfs.blocksize,
         }
 
+    @accepts(
+        Str('path'),
+        Bool('simplified', default=True),
+    )
+    def getacl(self, path, simplified=True):
+        """
+        Return ACL of a given path.
+        Simplified returns a shortened form of the ACL permset and flags
+        - TRAVERSE = sufficient rights to traverse a directory, but not read contents.
+        - READ = sufficient rights to traverse a directory, and read file contents.
+        - MODIFIY = sufficient rights to traverse, read, write, and modify a file. Equivalent to modify_set.
+        - FULL_CONTROL = all permissions.
+        - SPECIAL = does not fit into any of the above categories.
+
+        In all cases we replace USER_OBJ, GROUP_OBJ, and EVERYONE with owner@, group@, everyone@ for
+        consistency with getfacl and setfacl. If one of aforementioned special tags is used, 'id' must
+        be set to None.
+
+        An inheriting empty everyone@ ACE is appended to non-trivial ACLs in order to enforce Windows
+        expectations regarding permissions inheritance. This entry is removed from NT ACL returned
+        to SMB clients when 'ixnas' samba VFS module is enabled. We also remove it here to avoid confusion.
+        """
+        if not os.path.exists(path):
+            raise CallError('Path not found.', errno.ENOENT)
+
+        a = acl.ACL(file=path)
+        fs_acl = a.__getstate__()
+
+        if not simplified:
+            advanced_acl = []
+            for entry in fs_acl:
+                aclwho = acltools.ACLWho[entry['tag']]
+                ace = {
+                    'tag': aclwho.value,
+                    'id': entry['id'],
+                    'type': entry['type'],
+                    'perms': entry['perms'],
+                    'flags': entry['flags'],
+                }
+                if ace['tag'] == 'everyone@' and acltools.convert_to_basic_permset(ace['perms']) == 'NOPERMS':
+                    continue
+                advanced_acl.append(ace)
+            return advanced_acl
+
+        if simplified:
+            simple_acl = []
+            for entry in fs_acl:
+                aclwho = acltools.ACLWho[entry['tag']]
+                ace = {
+                    'tag': aclwho.value,
+                    'id': entry['id'],
+                    'type': entry['type'],
+                    'perms': acltools.convert_to_basic_permset(entry['perms']),
+                    'flags': acltools.convert_to_basic_flagset(entry['flags']),
+                }
+                if ace['tag'] == 'everyone@' and ace['perms'] == 'NOPERMS':
+                    continue
+                simple_acl.append(ace)
+
+            return simple_acl
+
+    @accepts(
+        Str('path'),
+        List('dacl', default=[]),
```

Comment:
I think we should have something like this in List dacl:

`items=[Dict('daclentry', Str('tag'), ...]`

Reply:
Done.